### DISCUSSION TITLE
[feature request] autolink references should apply to tag messages

### DISCUSSION THREAD

**Discussion by osievert-gpfw** [community/community#66356] [Repositories]
### Select Topic Area

Product Feedback

### Body

Github has a great feature [autolinked references](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/autolinked-references-and-urls) which in several pages on the github website automatically converts certain character strings to hyperlinks. We use that extensively to link a git commit to our issue tracker. Repository owners are able to set up their own autolink references, which is a critical piece of this solution.

Example: if in the first line of a git commit I put the message
```
fix: re-aligns the fruggleblast [KEWL-104]
```

then when I look at the pull request or the commit log for this on github.com, the KEWL-104 text is shown to me as a hyperlink to `https://my-issue-tracker.com/KEWL-104` because I set up a custom autolink reference for that. With this single breadcrumb, the reverse is true as well - our issue tracker automatically detects these and links back to the git commit. This is a huge productivity booster, because anyone doing code reviews or just browsing the repository can easily jump over to the issue tracker for more context about the change. Anyone who is browsing the issue tracker can easily jump over to the git repository to examine the code changes.

All fine and good. On to the feature request part: it would be even better if this same autolink reference treatment were given to the web page that displays git tag information. Git tags can be annotated with a message. We use this message to leave release notes about significant changes since the last release in a repository. These release notes display the exact same kinds of breadcrumbs, e.g. `[KEWL-104]`. Today, those are displayed as dead text in the tag webpage (e.g., `https://github.com/org/repo/releases/tag/2.5.7`). Because the autolinked references are not applied to this page, the tags and associated release notes are dead ends. Unlike other pages on github, such as the commit history page, the commit detail page, and the pull request page, these are not displayed as hyperlinks on the tag page. This means there is very little point in looking at this page - without these hyperlinks you cannot dig deeper to learn more.

In summary: please apply the autolink reference treatment to the web page that displays git tags (`https://github.com/<org>/<repo>/releases/tag/<tag>`).
